### PR TITLE
ci: version to 4.0.0 Micronaut 4.0.0-SNAPSHOT githubCoreBranch=4.0.x …

### PR DESCRIPTION
**Pull request by wetted**
…remove deprecations

### CODE REVIEW

**Comment by wetted** [comment]
This still has some broken tests to address, as well as #1091

File: security-oauth2/src/main/java/io/micronaut/security/oauth2/client/clientcredentials/propagation/ClientCredentialsHttpClientFilter.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ public ClientCredentialsHttpClientFilter(OutgoingHttpRequestProcessor outgoingHt
                                              BeanContext beanContext) {
         this.outgoingHttpRequestProcessor = outgoingHttpRequestProcessor;
         this.oauthClientConfigurationCollection = oauthClientConfigurationStream
-                .filter(config -> config.getClientCredentials().map(ClientCredentialsConfiguration::isEnabled).orElse(false))
```
Comment (sdelamo) [inline security-oauth2/src/main/java/io/micronaut/security/oauth2/client/clientcredentials/propagation/ClientCredentialsHttpClientFilter.java:79]: why this change?

**Comment by sdelamo** [comment]
@wetted re-request a review once the build passes. Let me know if you need help.

File: security-oauth2/src/main/java/io/micronaut/security/oauth2/client/clientcredentials/propagation/ClientCredentialsHttpClientFilter.java (hunk 1 above)
Comment (wetted) [reply security-oauth2/src/main/java/io/micronaut/security/oauth2/client/clientcredentials/propagation/ClientCredentialsHttpClientFilter.java:79]: That's an artifact of when I started looking into deprecating the method referred to by #1091. I didn't mean to push that change. I have reverted that for now.

**Review by sdelamo — APPROVED** [review]
